geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSUnionStoreIntegrationTest.java

```diff
@@ -0,0 +1,267 @@
+    assertThat(jedis.sunionstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2)).isEqualTo(result.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(result);
+
+  }
+
+  @Test
+  public void sunionstore_withNonExistentDest_withSomeSharedValues_returnsUnionSize_storesUnion() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    jedis.sadd(SET_KEY_2, secondSetMembers);
+
+    String[] result = {"one", "two", "three", "four", "five", "linux", "peach"};
+    assertThat(jedis.sunionstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2)).isEqualTo(result.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionstore_withNonExistentDest_withAllSharedValues_returnsUnionSize_storesUnion() {
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    jedis.sadd(SET_KEY_2, SET_MEMBERS_1);
+
+    assertThat(jedis.sunionstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2))
+        .isEqualTo(SET_MEMBERS_1.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(SET_MEMBERS_1);
+  }
+
+  @Test
+  public void sunionstore_withNonExistentDest_withMultipleNonExistentSets_returnsZero_destKeyDoesNotExist() {
+    assertThat(jedis.sunionstore(DESTINATION_KEY, NON_EXISTENT_SET, "{tag1}nonExistentSet2"))
+        .isEqualTo(0);
+    assertThat(jedis.exists(DESTINATION_KEY)).isFalse();
+  }
+
+  @Test
+  public void sunionstore_withExistentDest_withExistentSet_returnsUnionSize_storesUnion() {
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    assertThat(jedis.sunionstore(DESTINATION_KEY, SET_KEY_1)).isEqualTo(SET_MEMBERS_1.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(SET_MEMBERS_1);
+  }
+
+  @Test
+  public void sunionstore_withExistentDest_withNonExistentSet_returnsZero_destKeyDoesNotExist() {
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    assertThat(jedis.sunionstore(DESTINATION_KEY, NON_EXISTENT_SET)).isEqualTo(0);
+    assertThat(jedis.exists(DESTINATION_KEY)).isFalse();
+  }
+
+  @Test
+  public void sunionstore_withExistentDest_withNonOverlappingSets_returnsUnionSize_storesUnion() {
+    String[] secondSetMembers = new String[] {"apple", "microsoft", "linux", "peach"};
+    jedis.sadd(DESTINATION_KEY, DESTINATION_MEMBERS);
+    jedis.sadd(SET_KEY_1, SET_MEMBERS_1);
+    jedis.sadd(SET_KEY_2, secondSetMembers);
+
+    String[] result =
+        {"one", "two", "three", "four", "five", "apple", "microsoft", "linux", "peach"};
+    assertThat(jedis.sunionstore(DESTINATION_KEY, SET_KEY_1, SET_KEY_2)).isEqualTo(result.length);
+    assertThat(jedis.smembers(DESTINATION_KEY)).containsExactlyInAnyOrder(result);
+
+  }
+
+  @Test
+  public void sunionstore_withExistentDest_withAllSharedValues_returnsUnionSize_storesUnion() {
```

Comment:
These two tests aren't really necessary, as we don't expect that the presence or otherwise of the destination key should have any impact on us correctly calculating the union of the source sets, and we have other tests that already verify that that behaviour is correct. To keep things easier to maintain, they can be removed.